components/style/sharing/mod.rs

```diff
@@ -584,14 +525,7 @@ impl<E: TElement> StyleSharingCandidateCache<E> {
 
             match sharing_result {
                 Ok(shared_styles) => {
-                    // Yay, cache hit. Share the style.
-                    let child_cascade_requirement =
-                        target.accumulate_damage_when_sharing(shared_context,
```

Comment:
Right, with this patch se arrive to finish_restyle instead of early-returning, which does all this :)